File: app/src/main/java/co/za/takeabyte/gradalot2020/promotions/uimodel/UIModelPromotionItem.java
package co.za.takeabyte.gradalot2020.promotions.uimodel;

import androidx.annotation.DrawableRes;
import androidx.annotation.NonNull;

import co.za.takeabyte.gradalot2020.R;

public class UIModelPromotionItem {
    private @DrawableRes int imageRes = R.drawable.product_image;
    private double price = Double.MIN_VALUE;

    private String title = "";

    public void setImageRes(@DrawableRes int imageRes) {
        this.imageRes = imageRes;
    }

    public int getImageRes() {
        return imageRes;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public void setTitle(@NonNull String title) {
        this.title = title;
    }

    public String getTitle() {
        return title;
    }

    public String getDisplayPrice() {
        return Double.toString(price);
    }
}

File: app/src/main/java/co/za/takeabyte/gradalot2020/promotions/uimodel/UIModelPromotionItem.kt
package co.za.takeabyte.gradalot2020.promotions.uimodel

import androidx.annotation.DrawableRes
import co.za.takeabyte.gradalot2020.R

data class UIModelPromotionItem(
    @DrawableRes val imageRes: Int = R.drawable.product_image,
    val title: String = String(),
    val price: Double = Double.MIN_VALUE
) {
    fun getDisplayPrice() = price.toString()
}